LAPACK_CPP/src/lapack_cpp/rot_cpp.cpp

```diff
@@ -0,0 +1,67 @@
+#ifdef USE_FORTRAN_BLAS
+    #include <cassert>
+    #include <complex>
+    #include <type_traits>
+
+    #include "lapack_c.h"
+    #include "lapack_cpp/base.hpp"
+    #include "lapack_cpp/blas/rot.hpp"
+
+namespace lapack_cpp {
+
+/**
+ * Templated wrapper around c functions, will be instantiated for each type.
+ *
+ * @tparam T
+ * @param A
+ * @param B
+ * @param C
+ */
+template <typename T, typename TC, typename TS, typename idx_t>
+inline void rot_c_wrapper(const Vector<T, idx_t>& x,
```

Comment:
Have you considered overloading? The advantage of overloading at this point would be that the user can add his own overloads:
```c++
inline void rot_c_wrapper(const Vector<float, idx_t>& x, const Vector<float, idx_t>& y, TC c, TS s) {
    assert(x.size() == y.size());
    return lapack_c_srot(x.size(), x.ptr(), x.stride(), y.ptr(), y.stride(), c, s);
}
inline void rot_c_wrapper(const Vector<double, idx_t>& x, const Vector<double, idx_t>& y, TC c, TS s) {
    assert(x.size() == y.size());
    return lapack_c_drot(x.size(), x.ptr(), x.stride(), y.ptr(), y.stride(), c, s);
}
...
```

Reply:
`rot_c_wrapper` is not meant to be user callable. It is just a utility so we can call it when instantiating the function we actually want to overload: `rot`. It is a pattern I designed for gemm, where the logic is more complicated. I guess having this function at all does not make that much sense here.